packages/gatsby-source-shopify/src/query-builders/product-variants-query.ts

```diff
@@ -39,7 +43,7 @@ export class ProductVariantsQuery extends BulkQuery {
                       originalSrc
                       transformedSrc
                     }
-                    inventoryItem {
+                    inventoryItem @include(if: ${includeLocations}) {
```

Comment:
Ideally we'd also use gql variables instead of string interpolation but it looks like that'd be a bigger refactor outside the scope of this PR since the plugin is doing that in a few places and doesn't seem to account for variables when building queries.